source/Applications/Basic/Visualization/CaptureWritePCLVis3D/CaptureWritePCLVis3D.cpp

```diff
@@ -1,18 +1,121 @@
 /*
-This example shows how capture point clouds, with color, from the Zivid camera,
-save it to PCD file format, and visualize it.
+This example shows how capture point clouds, with color and with/without normals, from the Zivid camera,
+save it to PCD file, and visualize it.
 */
 
-#include <Zivid/Visualization/Visualizer.h>
 #include <Zivid/Zivid.h>
 
 #include <pcl/io/pcd_io.h>
 #include <pcl/point_types.h>
 #include <pcl/visualization/cloud_viewer.h>
 
 #include <iostream>
+#include <thread>
 
-int main()
+namespace
+{
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr convertToPCLPointCloud(const Zivid::Array2D<Zivid::PointXYZColorRGBA> &data)
+    {
+        auto pointCloud = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>();
+        pointCloud->width = data.width();
+        pointCloud->height = data.height();
+        pointCloud->is_dense = false;
+        pointCloud->points.resize(data.size());
+        for(size_t i = 0; i < data.size(); ++i)
+        {
+            pointCloud->points[i].x = data(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].y = data(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].z = data(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].r = data(i).color.r; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].g = data(i).color.g; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].b = data(i).color.b; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return pointCloud;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGBNormal>::Ptr convertToPCLPointCloudWithNormals(
+        const Zivid::Array2D<Zivid::PointXYZColorRGBA> &data,
+        const Zivid::Array2D<Zivid::NormalXYZ> &normals)
```

Comment:
Would be nice to use an optional for normals here, but cannot use it for c++11 that are in the sample, since optional is c+++ 17